bootloader/ubootenv/env_test.go

```diff
@@ -25,94 +25,113 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"strings"
 	"testing"
 
 	. "gopkg.in/check.v1"
 
 	"github.com/snapcore/snapd/bootloader/ubootenv"
+	"github.com/snapcore/snapd/testutil"
 )
 
 // Hook up check.v1 into the "go test" runner
 func Test(t *testing.T) { TestingT(t) }
 
-type uenvTestSuite struct {
+type uenvBaseSuite struct {
 	envFile string
 }
 
-var _ = Suite(&uenvTestSuite{})
+func (s *uenvBaseSuite) SetUpTest(c *C) {
+	s.envFile = filepath.Join(c.MkDir(), "uboot.env")
+}
+
+type uenvNativeTestSuite struct {
+	uenvBaseSuite
+}
+
+type uenvTextTestSuite struct {
+	uenvBaseSuite
+}
+
+var _ = Suite(&uenvNativeTestSuite{})
+var _ = Suite(&uenvTextTestSuite{})
 
-func (u *uenvTestSuite) SetUpTest(c *C) {
-	u.envFile = filepath.Join(c.MkDir(), "uboot.env")
+func (u *uenvNativeTestSuite) SetUpTest(c *C) {
+	u.uenvBaseSuite.SetUpTest(c)
 }
 
-func (u *uenvTestSuite) TestSetNoDuplicate(c *C) {
-	env, err := ubootenv.Create(u.envFile, 4096)
+func (u *uenvNativeTestSuite) TestSetNoDuplicate(c *C) {
+	env, err := ubootenv.Create(u.envFile, ubootenv.NativeFormat, 4096)
 	c.Assert(err, IsNil)
 	env.Set("foo", "bar")
 	env.Set("foo", "bar")
 	c.Assert(env.String(), Equals, "foo=bar\n")
 }
 
-func (u *uenvTestSuite) TestOpenEnv(c *C) {
-	env, err := ubootenv.Create(u.envFile, 4096)
+func (u *uenvNativeTestSuite) TestSanityNativeIsNotText(c *C) {
+	native := filepath.Join(c.MkDir(), "uboot.env")
+	nenv, err := ubootenv.Create(native, ubootenv.NativeFormat, 4096)
+	c.Assert(err, IsNil)
+	nenv.Set("foo", "bar")
+
+	text := filepath.Join(c.MkDir(), "boot.sel")
+	tenv, err := ubootenv.Create(text, ubootenv.TextFormat, 4096)
+	c.Assert(err, IsNil)
+	tenv.Set("foo", "bar")
+
+	c.Assert(tenv.Get("foo"), Equals, nenv.Get("foo"))
+
+	// the string's are the same
+	c.Assert(tenv.String(), Equals, nenv.String())
+
+	// but the file contents aren't
+	c.Assert(tenv.Save(), IsNil)
+	c.Assert(nenv.Save(), IsNil)
+	nbytes, err := ioutil.ReadFile(native)
+	c.Assert(err, IsNil)
+	tbytes, err := ioutil.ReadFile(text)
+	c.Assert(err, IsNil)
+	c.Assert(tbytes, Not(Equals), nbytes)
```

Comment:
we should probably check instead that the native sizes are what we expect for either,  and that the text one doesn't have bytes that are not low ascii

Reply:
okay added tests for this, but note that the test about the text not having low ascii or \0, etc. is in another test from the sanity test